docs/ug/report-config.yaml

```diff
@@ -0,0 +1,11 @@
+repoUrl: github.com/user/repo
+reportTitle: RepoSense Report
+authorDisplayName: Sample Author
+authorGithubId: Sample Author Github ID
+branches:
+  - name: Branch 1
+    blurb: Blurb 1
+  - name: Branch 2
+    blurb: Blurb 2
+startDate: 2020-01-01
+endDate: 9999-12-31
```

Comment:
See https://github.com/reposense/RepoSense/issues/2161#issuecomment-2022776952 for an example that seems intuitive from the user POV

Reply:
@damithc Just wanted to clarify, do you mean that we should include the configs within the markdown files itself, and have the backend parse the markdown files and then generate the reports, or that the overall order/style/keys in the YAML file should match what was used in the example provided?

Reply:
@asdfghjkxd I wasn't specifically using markdown either. It is just a custom format that makes sense from the user's point of view. I'm fine if something close can be achieved by YAML or some other commonly-used format, provided it makes sense from the user (i.e., the user shouldn't have to do a lot of extra work just to conform to the file format)

Reply:
@damithc I see, sure thing, we will try to design a user-friendly and convenient config file format and update this PR and parent issue with any new findings/improvements!